### PR TITLE
collation in auth handshake are 1-byte

### PR DESCRIPTION
**Pull request by dvilaverde**
Fix for https://github.com/go-mysql-org/go-mysql/issues/864

This adjustment reverts an earlier change that made the collation in the auth handshake 2 bytes. Now it follows the MySQL packet specification and is 1-byte. If the collation has a ID of > 255 then `SET NAMES ...` is called after establishing the connection.

### CODE REVIEW

File: .golangci.yml
Comment (dveeden) [inline .golangci.yml:1]: These don't look related to the rest of the PR?

**Review by dveeden — APPROVED** [review]

File: .golangci.yml
Comment (dveeden) [reply .golangci.yml:1]:
```
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused. 
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused. 
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused. 
WARN [linters_context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
```

Ok. So they are deprecated. Indeed good to remove them. Thanks.

**Review by lance6716 — APPROVED** [review]
